In [ ]:
pip install dtale

     |████████████████████████████████| 14.4MB 276kB/s 
     |████████████████████████████████| 51kB 5.8MB/s 
     |████████████████████████████████| 645kB 39.8MB/s 
     |████████████████████████████████| 71kB 8.2MB/s 
     |████████████████████████████████| 13.2MB 195kB/s 
     |████████████████████████████████| 1.8MB 49.4MB/s 
     |████████████████████████████████| 79.9MB 59kB/s 
     |████████████████████████████████| 51kB 4.8MB/s 
     |████████████████████████████████| 81kB 8.0MB/s 
     |████████████████████████████████| 194kB 49.4MB/s 
     |████████████████████████████████| 358kB 42.2MB/s 
     |████████████████████████████████| 1.0MB 38.4MB/s 
     |████████████████████████████████| 3.5MB 20.8MB/s 
     |████████████████████████████████| 92kB 9.0MB/s 
     |████████████████████████████████| 1.8MB 34.8MB/s 
  Created wheel for ppscore: filename=ppscore-1.2.0-py2.py3-none-any.whl size=13050 sha256=25f4156cbb1188f8a146fa362d4a2d6b2c3f2691b52c082f8fa42db9c9bb6c0d
  Stored in dir

In [ ]:
import seaborn as sns
import pandas as pd

In [ ]:
#importing the dataset and exploring it
from google.colab import files
uploaded = files.upload()

Saving may18_publish.csv to may18_publish (2).csv


In [ ]:
df =pd.read_csv("may18_publish.csv")

In [ ]:
df
df.head

<bound method NDFrame.head of           JIO  4G  download    53484   -62         UP East
0         JIO  4G    upload   4503.0   -62         UP East
1        IDEA  4G  download  11162.0  -111         Gujarat
2        IDEA  4G  download   5311.0  -110         Gujarat
3        IDEA  4G  download   9014.0  -105         Gujarat
4        IDEA  4G  download   4730.0  -111         Gujarat
...       ...  ..       ...      ...   ...             ...
2317212   JIO  4G  download  54259.0   -64     West Bengal
2317213  IDEA  4G  download    986.0    na  Andhra Pradesh
2317214  IDEA  4G  download   6528.0    na  Andhra Pradesh
2317215  IDEA  4G  download  11150.0    na  Andhra Pradesh
2317216  IDEA  4G  download   4185.0    na  Andhra Pradesh

[2317217 rows x 6 columns]>

NOTE: There was a problem in reading the file so here 5384 means Data Speed and -62 means Signal Strength 

In [ ]:
import dtale

In [ ]:
dtale.show(df)

2021-05-10 15:18:32,727 - INFO     - NumExpr defaulting to 2 threads.


http://966e892834f3:40000/dtale/main/1

In [ ]:
import dtale
import dtale.app as dtale_app

dtale_app.USE_COLAB = True

dtale.show(df)

https://80xvqgq74ne-496ff2e9c6d22116-40000-colab.googleusercontent.com/dtale/main/2

In [ ]:
df.head

<bound method NDFrame.head of            JIO  4G  download    53484   -62         UP East
0          JIO  4G    upload   4503.0   -62         UP East
1         IDEA  4G  download  11162.0  -111         Gujarat
2         IDEA  4G  download   5311.0  -110         Gujarat
3         IDEA  4G  download   9014.0  -105         Gujarat
4         IDEA  4G  download   4730.0  -111         Gujarat
...        ...  ..       ...      ...   ...             ...
401364  AIRTEL  3G  download   2291.0   -85  Madhya Pradesh
401365  AIRTEL  3G  download   7105.0   -69  Madhya Pradesh
401366  AIRTEL  3G  download    356.0   -69  Madhya Pradesh
401367  AIRTEL  3G  download    542.0   -85  Madhya Pradesh
401368  AIRTEL  3G       NaN      NaN   NaN             NaN

[401369 rows x 6 columns]>

In [ ]:

import numpy as np
import pandas as pd
import plotly.graph_objs as go

if isinstance(df, (pd.DatetimeIndex, pd.MultiIndex)):
	df = df.to_frame(index=False)


df = df.reset_index().drop('index', axis=1, errors='ignore')
df.columns = [str(c) for c in df.columns]  # update columns to strings in case they are numbers

s = df[~pd.isnull(df['UP East'])]['UP East']
chart = pd.value_counts(s).to_frame(name='data')
ordinal_data = df.groupby('UP East')[['53484']].sum()
chart['ordinal'] = ordinal_data
chart.index.name = 'labels'
chart = chart.reset_index().sort_values('ordinal')
chart = chart[:100]
charts = [go.Bar(x=chart['labels'].values, y=chart['data'].values, name='Frequency')]
charts.append(go.Scatter(
	x=chart['labels'].values, y=chart['ordinal'].values, yaxis='y2',
	name='53484 (sum)', line={'shape': 'spline', 'smoothing': 0.3}, mode='lines'
))
figure = go.Figure(data=charts, layout=go.Layout({
    'barmode': 'group',
    'legend': {'orientation': 'h'},
    'title': {'text': 'UP East Value Counts'},
    'xaxis': {'title': {'text': 'UP East'}},
    'yaxis': {'title': {'text': 'Frequency'}},
    'yaxis2': {'overlaying': 'y', 'side': 'right', 'title': {'text': '53484 (sum)'}}
}))
import plotly.io as pio
pio.renderers.default = 'colab'

In [ ]:
figure

Inferences:

Total Rows: 401,369

Top performering State of may 2018: UP East

In [ ]:


import numpy as np
import pandas as pd
import plotly.graph_objs as go

if isinstance(df, (pd.DatetimeIndex, pd.MultiIndex)):
	df = df.to_frame(index=False)


df = df.reset_index().drop('index', axis=1, errors='ignore')
df.columns = [str(c) for c in df.columns]  # update columns to strings in case they are numbers

s = df[~pd.isnull(df['download'])]['download']
chart = pd.value_counts(s).to_frame(name='data')
ordinal_data = df.groupby('download')[['53484']].sum()
chart['ordinal'] = ordinal_data
chart.index.name = 'labels'
chart = chart.reset_index().sort_values('ordinal')
chart = chart[:100]
charts = [go.Bar(x=chart['labels'].values, y=chart['data'].values, name='Frequency')]
charts.append(go.Scatter(
	x=chart['labels'].values, y=chart['ordinal'].values, yaxis='y2',
	name='53484 (sum)', line={'shape': 'spline', 'smoothing': 0.3}, mode='lines'
))
figure = go.Figure(data=charts, layout=go.Layout({
    'barmode': 'group',
    'legend': {'orientation': 'h'},
    'title': {'text': 'download Value Counts'},
    'xaxis': {'title': {'text': 'download'}},
    'yaxis': {'title': {'text': 'Frequency'}},
    'yaxis2': {'overlaying': 'y', 'side': 'right', 'title': {'text': '53484 (sum)'}}
}))


In [ ]:
figure

Inference:
For the month of May 2018
So from this we can infer that 
Upload Data Speed (5.170B) and Download Data Speed(16.90B) and their Frequency is almost Similar.


In [ ]:


import numpy as np
import pandas as pd
import plotly.graph_objs as go

if isinstance(df, (pd.DatetimeIndex, pd.MultiIndex)):
	df = df.to_frame(index=False)


df = df.reset_index().drop('index', axis=1, errors='ignore')
df.columns = [str(c) for c in df.columns]  # update columns to strings in case they are numbers

s = df[~pd.isnull(df['53484'])][['53484']]
chart, labels = np.histogram(s, bins=20)
import scipy.stats as sts
kde = sts.gaussian_kde(s['53484'])
kde_data = kde.pdf(np.linspace(labels.min(), labels.max()))
# main statistics
stats = df['53484'].describe().to_frame().T
charts = [
	go.Bar(x=labels[1:], y=chart, name='Histogram'),
	go.Scatter(
		x=list(range(len(kde_data))), y=kde_data, name='KDE',		yaxis='y2', xaxis='x2',		line={'shape': 'spline', 'smoothing': 0.3}, mode='lines'
	)
]
figure = go.Figure(data=charts, layout=go.Layout({
    'barmode': 'group',
    'legend': {'orientation': 'h'},
    'title': {'text': '53484 Histogram (bins: 20) w/ KDE'},
    'xaxis2': {'anchor': 'y', 'overlaying': 'x', 'side': 'top'},
    'yaxis': {'side': 'left', 'title': {'text': 'Frequency'}},
    'yaxis2': {'overlaying': 'y', 'side': 'right', 'title': {'text': 'KDE'}}
}))
import plotly.io as pio
pio.renderers.default = 'colab'


In [ ]:
figure

Maximum Frequency of occurence is of Signal Strength of 6913.1 Mbps with a Frequency is 1.4485M and Data Speed of 78.87ubps.

Insights from the Month of May 2018

Top Performing State in terms of Value Counts was Up East Division with a 
frequency of about 33.246k users on average and least from Himachal
Total Rows: 401,369



Upload Data Speed (5.170B) and Download Data Speed(16.90B) and their Frequency is almost Similar.

Maximum Frequency of occurence is of Signal Strength of 6913.1 Mbps with a Frequency is 1.4485M and Data Speed of 78.87ubps.